libs/langchain-azure-openai/src/chat_models.ts

```diff
@@ -256,7 +256,8 @@ export class AzureChatOpenAI
       getEnvironmentVariable("AZURE_OPENAI_API_ENDPOINT");
 
     this.azureOpenAIApiCompletionsDeploymentName =
-      fields?.azureOpenAIEmbeddingsApiDeploymentName ??
+      (fields?.azureOpenAIEmbeddingsApiDeploymentName ||
```

Comment:
Should this even take into account `azureOpenAIEmbeddingsApiDeploymentName`? I guess we have to leave it there for backwards compatibility...

Reply:
While backward compatibility is important, I'm leaning towards removing azureOpenAIEmbeddingsApiDeploymentName since it seems to confuse users.
What do you think?

Reply:
I created a PR for deprecating `azureOpenAIEmbeddingsApiDeploymentName` .
https://github.com/langchain-ai/langchainjs/pull/4714